lib/trie/test_utils.go

```diff
@@ -39,14 +40,17 @@ func GenerateRandomTests(t testing.TB, size int) []Test {
 }
 
 func generateRandomTest(t testing.TB, kv map[string][]byte) Test {
-	r := *rand.New(rand.NewSource(rand.Int63())) //nolint
 	test := Test{}
 
 	for {
 		n := 2 // arbitrary positive number
-		size := r.Intn(510) + n
-		buf := make([]byte, size)
-		_, err := r.Read(buf)
+		size, err := rand.Int(rand.Reader, big.NewInt(510))
+		if err != nil {
```

Comment:
`require.NoError(t, err)`
Also, replace in other places.

Reply:
updated.